java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -328,6 +330,11 @@ class DatabaseImpl(
                 PrimitiveType.Char.id -> it.getNullableInt(3)?.toChar()?.toReferencable()
                 PrimitiveType.Float.id -> it.getNullableFloat(5)?.toReferencable()
                 PrimitiveType.Double.id -> it.getNullableDouble(5)?.toReferencable()
+                PrimitiveType.BigInt.id -> if (it.isNull(4)) {
+                    null
+                } else {
+                    BigInteger(it.getString(4)).toReferencable()
+                }
```

Comment:
I think this can be simplified eg:
PrimitiveType.BigInt.id -> it.getNullableString(4)?.let {s -> BigInteger(s).toReferencable() }

Reply:
Really nice

Reply:
I don't think that's simpler.. what's currently there matches what's used in the else -> clause too.